### PR TITLE
ft(pipeline): Add variables flag for `pipeline run`

### LINKED ISSUE #319: Pass variables when running a pipeline
When running pipelines we use variables to trigger a release to the specified environment.

An additional argument when calling `glab pipeline run` such as `--variables KEY:VALUE` would be ideal.

Currently, the only alternative is to use the web UI.

### PR DESCRIPTION
**Pull request by zemzale**
**Description**
This adds flag for running pipelines with custom variables passed to it.
They all default to `env_var`, for simplicity.

**Related Issue**

#319

**Motivation and Context**

This allows to run pipelines from command line with custom variables. 
**How Has This Been Tested?**

**Screenshots (if appropriate):**
image

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Checklist:**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by zemzale** [comment]
@profclems I wrote up this MR so we can add this feature. Look over it please comment if there are some problems you see. 
I intend to add some tests for this but I don't have the time for it right now, but would love some feedback so I can know that this is going the right way.

File: commands/pipeline/run/pipeline_run.go [hunk 1]
```diff
@@ -57,17 +63,30 @@ func NewCmdRun(f *cmdutils.Factory) *cobra.Command {
...
 
-			// TODO: support setting pipeline variables via cli.
-			v := []*gitlab.PipelineVariable{
+			pipelineVars := []*gitlab.PipelineVariable{
 				{
 					Key:          "GLAB_CLI_KEY",
 					Value:        "GLAB_CLI_VAL",
 					VariableType: "env_var",
 				},
 			}
```
Comment (profclems) [inline commands/pipeline/run/pipeline_run.go]: I don't think this should be there anymore. You can initialize as an empty slice

```suggestion
			pipelineVars := []*gitlab.PipelineVariable{}
```

**Review by profclems — CHANGES_REQUESTED** [review]
I'm okay with the implementation. Looks clean 🎉

**Comment by profclems** [comment]
> @profclems I wrote up this MR so we can add this feature. Look over it please comment if there are some problems you see.
> I intend to add some tests for this but I don't have the time for it right now, but would love some feedback so I can know that this is going the right way.

I actually like the implementation. This looks good. You can add the tests in a separate PR when you're ready

File: commands/pipeline/run/pipeline_run.go (hunk 1 above)
Comment (zemzale) [reply commands/pipeline/run/pipeline_run.go]: Okay, didn't know if this is needed so just left it there. Now is removed

**Comment by zemzale** [comment]
Yeah looks like I might add tests later on. I might not have the time in the next days, and never had done testing for CLIs like this so it's a bit to wrap my head around. If everything is okay then you are free to merge this

**Review by profclems — APPROVED** [review]

**Pull request merged**